asset-transfer-private-data/application-javascript/app.js

```
@@ -0,0 +1,273 @@
+    const caService = new FabricCAServices(caInfo.url, { trustedRoots: caTLSCACerts, verify: false }, caInfo.caName);
+
+    await caUtils.EnrollOrgAdminUser(mspOrg2, walletOrg2, caService);
+
+    // register & enroll application user with CA, which is used as client identify to make chaincode calls, stores app user identity in local wallet
+    await caUtils.RegisterOrgUser(caUtils.Org2UserId, mspOrg2, walletOrg2, caService);
+
+    try {
+        // Create a new gateway for connecting to Org's peer node.
+        const gatewayOrg2 = new Gateway();
+        await gatewayOrg2.connect(ccpOrg2,
+            { wallet: walletOrg2, identity: caUtils.Org2UserId, discovery: { enabled: true, asLocalhost: true } });
+
+        return gatewayOrg2;
+    } catch (error) {
+        console.error(`Error in connecting to gateway: ${error}`);
+        process.exit(1);
+    }
+}
+
+// Main workflow : usecase details at asset-transfer-private-data/chaincode-go/README.md
+// This app uses fabric-samples/test-network based setup and the companion chaincode
+// For this usecase illustration, we will use both Org1 & Org2 client identity from this same app
+// In real world the Org1 & Org2 identity will be used in different apps to achieve asset transfer.
+async function main() {
+    try {
+
+        /** ******* Fabric client init: Using Org1 identity to Org1 Peer ********** */
+        const gatewayOrg1 = await initContractFromOrg1Identity();
+        const networkOrg1 = await gatewayOrg1.getNetwork(myChannel);
+        const contractOrg1 = networkOrg1.getContract(myChaincodeName);
+        // Since this sample chaincode uses, Private Data Collection level endorsement policy, addDiscoveryInterest
+        // scopes the discovery service further to use the endorsement policies of collections, if any
+        contractOrg1.addDiscoveryInterest({ name: myChaincodeName, collectionNames: [memberAssetCollectionName, org1PrivateCollectionName] });
```

@btl5037 thanks Brett !  addDiscoveryInterest scoping worked in selecting own orgs endorsing peer. This api suppresses the error seen in sdk , when I was using setEndorsingOrganizations at each Txn level.

---

Awesome glad it worked. Sorry for steering you wrong in the first place, but we both learned something.